### PR TITLE
Implement [M3C] Aurora Shifter

### PR DESCRIPTION
**Pull request by grimreap124**
[[Aurora Shifter]]

### CODE REVIEW

**Review by Susucre — COMMENTED** [review]
I would appreciate unit testing for this one.
Giving ability in Copy effects are usually a little subtle and it is best if they stay working in the long run.

**Comment by grimreap124** [comment]
> > I would appreciate unit testing for this one. Giving ability in Copy effects are usually a little subtle and it is best if they stay working in the long run.
> 
> I'll add one soon

Maybe not soon but it's added now

File: Mage.Sets/src/mage/cards/a/AuroraShifter.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public AuroraShifter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.CREATURE }, "{1}{U}");
+
+        this.subtype.add(SubType.SHAPESHIFTER);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Whenever Aurora Shifter deals combat damage to a player, you get that many {E}.
+        this.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
+                new GetEnergyCountersControllerEffect(SavedDamageValue.MUCH)));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/a/AuroraShifter.java:43]: MANY instead of MUCH

File: Mage.Sets/src/mage/cards/a/AuroraShifter.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        return false;
+    }
+}
+
+class AuroraShifterCopyApplier extends CopyApplier {
+
+    @Override
+    public boolean apply(Game game, MageObject blueprint, Ability source, UUID copyToObjectId) {
+
+        blueprint.getAbilities().add(new DealsCombatDamageToAPlayerTriggeredAbility(
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/a/AuroraShifter.java:102]: Don't you need to also add the primary ability? Like Cryptoplasm.

File: Mage.Tests/src/test/java/org/mage/test/cards/single/mh3/AuroraShifterTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, shifter);
+        addCard(Zone.BATTLEFIELD, playerA, "Grizzly Bears");
+
+//        Combat 1
+        attack(1, playerA, shifter, playerB);
+        runCode("energy counter is 1", 1, PhaseStep.POSTCOMBAT_MAIN, playerA, (info, player, game) -> checkEnergyCount(info, player, 1));
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/single/mh3/AuroraShifterTest.java:41]: don't call execute more than once in a test - only at the end, use checkXXX if you need interim asserts

**Review by xenohedron — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/a/AuroraShifter.java (hunk 2 above)
Comment (grimreap124) [reply Mage.Sets/src/mage/cards/a/AuroraShifter.java:102]: No  I don't believe it has the transform ability on the creature it transforms into just the energy gain for damage amount

**Review by xenohedron — APPROVED** [review]

**Pull request merged**